server.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const net = require('net');
+const server = net.createServer();
+const PORT = process.env.PORT || 3000;
+const Client = require(`${__dirname}/model/client.js`);
+const EE = require('events');
+const ee = new EE();
+
+const chatGroup = [];
+
+///+++++ socket events ++++++\\\
+
+ee.on('login', function(socket) {
+  const client = new Client(socket);
+  chatGroup.push(client);
+  ee.emit('message', socket, client);
+  socket.on('error', (error) => {
+    ee.emit('socketError', error);
+  });
+  socket.on('close', (socket) => {
+    ee.emit('logout', socket, client);
+  });
+});
+
+ee.on('socketError', (error) => {
+  if (error) console.error(error);
+});
+
+ee.on('logout', (socket, client) => {
+  chatGroup.forEach((ele, index) => {
+    if (ele.id === client.id) {
+      // let removeUser = chatGroup.indexOf(client);
+      chatGroup.splice(index, 1);
+    }
+  });
+});
+
+///+++++ Message from client pool sockets handling +++++\\\
+
+ee.on('message', (socket, client) => {
+  socket.on('data', function(data) {
+    let userTarget;
+    let directMsg;
+    if (data.toString().split(' ').length > 2) {
+      userTarget = data.toString().split(' ').slice(1).shift().trim();
+      directMsg = data.toString().split(' ').slice(2).join(' ');
+    }
+    const command = data.toString().split(' ').shift().trim();
+    const msg = data.toString().split(' ').slice(1).join(' ');
+
+    if(command.includes('@all')) {
```

Comment:
you have a lot of repeated code here. You could probably D.R.Y. this out and use just one conditional. 
``` javascript
if(command.startsWith('@')) {
      ee.emit(command, client, data.toString().split(' ').slice(1).join(' '));
      return;
    }
```

JavaScript allows this, you can pass any arbitrary number of arguments to a function so you don't necessarily need to create a different conditional statement to handle each command.